pkg/client/unversioned/portforward/portforward_test.go

```diff
@@ -218,33 +220,23 @@ var _ kubelet.PortForwarder = &fakePortForwarder{}
 func (pf *fakePortForwarder) PortForward(name string, uid types.UID, port uint16, stream io.ReadWriteCloser) error {
 	defer stream.Close()
 
-	var wg sync.WaitGroup
-
-	// client -> server
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-
-		// copy from stream into a buffer
-		received := new(bytes.Buffer)
-		io.Copy(received, stream)
-
-		// store the received content
-		pf.lock.Lock()
-		pf.received[port] = received.String()
-		pf.lock.Unlock()
-	}()
-
-	// server -> client
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+	// read from the client
+	received := make([]byte, len(pf.expected[port]))
+	n, err := stream.Read(received)
```

Comment:
Is this supposed to test the case where the client sends more than expected? io.Copy doesn't just stop at len(received) like Read...